update

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+set -ex
+
+BRANCH=${1:-master}
+
+git clean -f -d
+GENERATED_PATHS=$(ls -1)
+
+# checkout source branch
+git checkout "$BRANCH" && git submodule update --init
+COPY_PATHS=$(ls -1)
+tag=$(git rev-parse HEAD)
+
+# copy the branch content into source/
+mkdir source
+for i in $COPY_PATHS; do
+    cp -r "$i" source
+done
+VERSIONS=$(sed -n 's/^VERSIONS[[:space:]]*=//p' source/Makefile)
+
+git checkout generated
+# Sometimes git does not remove common/ on branch change
+[ -d common ] && rm -rf common
+
+for i in $GENERATED_PATHS; do
+    case "$i" in
+    # Do not remove the update script itself
+    update)
+        continue
+        ;;
+    *)
+        rm -rf "$i"
+        ;;
+    esac
+done
+
+# Generate the sources inside source/ and copy them to root
+(
+    cd source
+    make generate-all
+    for i in $VERSIONS; do
+        cp -r "$i" ../
+    done
+)
+rm -rf source
+
+# Add both existing and deleted files to the index
+git add ./*
```

Comment:
This shouldn't have an effect for deleted files, the list of files is defined by shell path expansion.

Reply:
Yes, this adds all the files currently in the root to the index. The command to add deleted ones is below this line.

However, I guess it would be better to just add the content of $VERSIONS here, since that is all we care about.